### PR TITLE
Add generator mode to TF applier

### PR DESCRIPTION
**Pull request by henryre**
Add a method to TFApplier that takes a batch size and yields lists/DataFrames of transformed examples for batches. Implemented as separate `apply` method for stronger typing and for alignment with e.g. https://keras.io/preprocessing/image/.

**Test plan**
Added unit tests

### CODE REVIEW

File: test/augmentation/apply/test_tf_applier.py [hunk 1]
```diff
@@ -94,6 +99,53 @@ def test_tf_applier_keep_original_modify_in_place(self):
...
+    def test_tf_applier_returns_none_generator(self) -> None:
+        data = self._get_x_namespace()
+        applier = TFApplier([square_returns_none], policy, k=2, keep_original=True)
+        batches_expected = [[1, 1, 1, 2], [3, 81, 81]]
+        gen = applier.apply_generator(data, batch_size=2)
+        for batch, batch_expected in zip(gen, batches_expected):
+            self.assertEqual([x.num for x in batch], batch_expected)
+        self.assertEqual([x.num for x in data], DATA)
+
+    def test_tf_applier_keep_original_modify_in_place_generator(self) -> None:
```
Comment (bhancock8) [inline test/augmentation/apply/test_tf_applier.py]: It's a little hard to follow what this test is doing, both since "modify in place _and_ keep original" is a bit of a paradox, and because both the assert test below and the DATA_IN_PLACE_EXPECTED are generated programmatically, so for a user to understand the test they need to hold a few things in their heads at once.

Comment (henryre) [reply test/augmentation/apply/test_tf_applier.py]: Ok sg

**Comment by henryre** [comment]
@paroma yeah that's the plan. was holding off for now since parallel data aug isn't really expected by most users (esp. for generator mode). Will explicitly add to roadmap though.
